lib/spack/spack/util/module_cmd.py

```diff
@@ -29,9 +31,15 @@ def module(*args):
         tty.warn('module function operating in test mode')
         module_cmd = ". %s 2>&1" % args[1]
     if args[0] in module_change_commands:
+        os.environ['SPACK_LD_LIBRARY_PATH'] = spack.main.spack_ld_library_path
+
         # Do the module manipulation, then output the environment in JSON
         # and read the JSON back in the parent process to update os.environ
-        module_cmd += ' >/dev/null; python -c %s' % py_cmd
+        module_cmd += ' >/dev/null;'
```

Comment:
The shenanigans we came up with here deserve some comments.  Specifically this should explain `SPACK_NEW_LD_LIBRARY_PATH`, `SPACK_LD_LIBRARY_PATH`, and in general what is going on.  Probably worth breaking the whole python `sys.executable` execution bit into its own function.